libsolidity/ast/Types.cpp

```diff
@@ -2237,24 +2237,19 @@ bool TupleType::isImplicitlyConvertibleTo(Type const& _other) const
 		size_t minNumValues = targets.size();
 		if (!targets.back() || !targets.front())
 			--minNumValues; // wildcards can also match 0 components
-		if (components().size() < minNumValues)
-			return false;
-		if (components().size() > targets.size() && targets.front() && targets.back())
+		if ((components().size() < minNumValues) || (components().size() > targets.size() && targets.front() && targets.back()))
```

Comment:
This is too long for github diff view.

Reply:
Alternative notation is
```
if (
  (components().size() < minNumValues) ||
  (components().size() > targets.size() && targets.front() && targets.back())
)
  return false;
```

Though I think out of all these options the original one is better.